@here/harp-datasource-protocol/lib/InterpolatedProperty.ts

```diff
@@ -103,3 +67,170 @@ export function isInterpolatedProperty<T>(p: any): p is InterpolatedProperty<T>
+        }
+    } else if (property._stringEncodedNumeralType !== undefined) {
+        if (
+            property._stringEncodedNumeralType === StringEncodedNumeral.Meters ||
+            property._stringEncodedNumeralType === StringEncodedNumeral.Pixels
+        ) {
+            return getInterpolatedLength(property, level, pixelToMeters);
+        } else {
+            return getInterpolatedColor(property, level);
+        }
+    }
+    return getInterpolatedLength(property, level, pixelToMeters);
+}
+
+function getInterpolatedLength(
+    property: InterpolatedProperty<length>,
+    level: number,
+    pixelToMeters: number
+): number {
+    const nChannels = property.values.length / property.zoomLevels.length;
+    const interpolant = new interpolants[property.interpolationMode](
+        property.zoomLevels,
+        property.values,
+        nChannels
+    );
+    if (
+        property.interpolationMode === InterpolationMode.Exponential &&
+        property.exponent !== undefined
+    ) {
+        (interpolant as ExponentialInterpolant).exponent = property.exponent;
+    }
+    interpolant.evaluate(level);
+
+    if (property._metricUnitMask === undefined) {
+        return interpolant.resultBuffer[0];
+    } else {
+        const maskInterpolant = new interpolants[property.interpolationMode](
+            property.zoomLevels,
+            property._metricUnitMask,
+            1
+        );
+        if (
+            property.interpolationMode === InterpolationMode.Exponential &&
+            property.exponent !== undefined
+        ) {
+            (maskInterpolant as ExponentialInterpolant).exponent = property.exponent;
+        }
+        maskInterpolant.evaluate(level);
+
+        return (
+            interpolant.resultBuffer[0] * (1 - maskInterpolant.resultBuffer[0]) +
+            interpolant.resultBuffer[0] * pixelToMeters * maskInterpolant.resultBuffer[0]
+        );
+    }
+}
+
+function getInterpolatedColor(property: InterpolatedProperty<color>, level: number): number {
+    const nChannels = property.values.length / property.zoomLevels.length;
+    const interpolant = new interpolants[property.interpolationMode](
+        property.zoomLevels,
+        property.values,
+        nChannels
+    );
+    if (
+        property.interpolationMode === InterpolationMode.Exponential &&
+        property.exponent !== undefined
+    ) {
+        (interpolant as ExponentialInterpolant).exponent = property.exponent;
+    }
+    interpolant.evaluate(level);
+
+    const rgbValues = hslToRGB(
+        interpolant.resultBuffer[0],
+        interpolant.resultBuffer[1],
+        interpolant.resultBuffer[2]
+    );
+    let result = "0x";
+    for (const value of rgbValues) {
+        // tslint:disable:no-bitwise
+        result += ("0" + Math.round(MathUtils.clamp(value, 0, 1) * 255).toString(16)).slice(-2);
+    }
+    return parseInt(result, 16);
+}
+
+function hslToRGB(h: number, s: number, l: number): number[] {
```

Comment:
Why we can't use THREE.Color here?

Reply:
All of these functions operate with component values, as the interpolation buffers don't use THREE.Color until the interpolation has been performed.

Reply:
```new THREE.Color().setHSL(h, s, l).toArray()``` ?

Reply:
I mean, we already using three.js and it has alot of color math, why not to use the code we already have?

Reply:
Fair point, will use this for consistency.